source/simulator/assembly.cc

```diff
@@ -1227,6 +1227,206 @@ namespace aspect
         }
     };
 
+    /**
+     * A class for the definition of functions that implement the
+     * non-linear system terms for the newton compressible or
+     * incompressible equations.
+     */
+    template <int dim>
+    class NewtonSystem : public aspect::internal::Assembly::Assemblers::AssemblerBase<dim>,
```

Comment:
call it `NewtonStokes` or similar to make clear that you're only applying the Newton solver to this part of the problem.